src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,390 @@
+  }
+
+  public double getRBearing() {
+    //Use law of sines to find the bearing of the arm's resultant vector, in radians
+    return Math.asin((kElbowArmLength * Math.sin(getElbowAngle())) / getRMagnitude()) + (Math.PI - getBaseAngle());
+  }
+
+  public double getArmX() {
+    return -1 * (getRMagnitude() * Math.cos(getRBearing()));
+  }
+
+  public double getArmY() {
+    return getRMagnitude() * Math.sin(getRBearing());
+  }
+
+  //This updates the arm positions to be used for stuff like shuffleboard
+  public void calculateCurrentPositions() {
+    armXPosition = getArmX();
+    armYPosition = getArmY();
+  }
+  
+
+  public void updateMotors() {
+    baseMotor.set(MathUtil.clamp(basePIDController.calculate(getBaseAngle()),-1,1));
+    elbowMotor.set(MathUtil.clamp(elbowPIDController.calculate(getElbowAngle()),0,1));
+  }
+
+
+  public void updateSimMotors() {
+    updateMotors();
+    baseArmSim.setInputVoltage(baseMotor.get() * RobotController.getBatteryVoltage());
+    elbowArmSim.setInputVoltage(elbowMotor.get() * RobotController.getBatteryVoltage());
+  }
+
+  //Angle finding methods
+  //The documentation for these can be found at our notion page at
+  //https://www.notion.so/2158/Arm-System-0fc38426812e42f2944f34668059c2e4
+  public static double convertToPrelimAngle(double x, double y) {
+    return (
+      -1*Math.acos(
+        ( (x*x) + (y*y) - (kBaseArmLength*kBaseArmLength) - (kElbowArmLength*kElbowArmLength) )/(2*kBaseArmLength*kElbowArmLength)
+      )
+    );
+  }
+
+  public static double convertToBaseAngle(double x, double y) {
+    double pAngle = convertToPrelimAngle(x, y);
+    return (
+      Math.atan(y/x)-Math.atan((kElbowArmLength*Math.sin(pAngle) )/( (kBaseArmLength+(kElbowArmLength*Math.cos(pAngle))))
+      )
+    );
+  }
+
+  public static double convertToElbowAngle(double x, double y) {
+    return Units.degreesToRadians(180) + convertToPrelimAngle(x, y);
+  }
+
+  public boolean atSetpoint(){
+    return basePIDController.atGoal() && elbowPIDController.atGoal();
+  }
+
+  public void setState(ArmState state){
+    double desiredBaseAngle = convertToBaseAngle(state.getX(),state.getY());
+    double desiredElbowAngle = convertToElbowAngle(state.getX(),state.getY());
+    desiredXPosition.setDouble(state.getX());
+    desiredYPosition.setDouble(state.getY());
+    currentState.setString(state.toString());
+
+    basePIDController.setTolerance(state == ArmState.TRANSITION ? 10 : 1);
+    elbowPIDController.setTolerance(state == ArmState.TRANSITION ? 10 : 1);
+
+    setBaseReference(desiredBaseAngle);
+    setElbowReference(desiredElbowAngle);
+  }
+  
+  public Command goToState(ArmState state){
+    //Command for autonomous, obstructs routine until arm is at setpoint
+    return new InstantCommand(() -> setState(state), this)
+      .andThen(new RepeatCommand(new InstantCommand(this::updateMotors, this))
+      .until(this::atSetpoint));
+  }
```

Comment:
```suggestion
  public Command goToState(ArmState state){
    //Command for autonomous, obstructs routine until arm is at setpoint
    return new FunctionalCommand(
                () -> setState(state), //Init
                this::updateMotors, //Execute
                (b)->{}, //End 
                this::atSetpoint, //isFinished
                this
            );
  }
```
You're using a command sequence to do what a single Command can do on its own.